lib/getHashDigest.js

```diff
@@ -36,15 +43,35 @@ function encodeBufferToBase(buffer, base) {
 	return output;
 }
 
+function getBase(digestType) {
+	return digestType === BASE_EMOJI ? emojiList.length : digestType.substr(4);
+}
+
+function sliceEmojis(output, maxLength) {
+	let match = emojiRegex.exec(output);
+	const emojis = [];
+	while(match) {
+		emojis.push(match[0]);
+		match = emojiRegex.exec(output);
+	}
+	return emojis.splice(0, maxLength).join("");
+}
+
+function sliceOutput(output, digestType, maxLength) {
```

Comment:
To simpily this could you:

* make `encodeBufferToBase` return an array.
* Use `Array.prototype.slice` instead of `String.prototype.substring` to ensure `maxLength`
* Convert array to string with join after that.
* Remove sliceEmojis, sliceOutput, emoji-regex

Reply:
As for the `emoji-regex` package. I'd encourage to keep it as it is better in detecting the different code points of emojis. Here is an example https://runkit.com/kohlikohl/emoji-regex.
The simple regex seems to only detect emojis that have two code points. Flags for example have four so it would miss identify and split them.
Maybe it is not relevant in this case, if so let me know and I'll make the final change.